src/swap/saga.ts

```diff
@@ -349,14 +349,14 @@ function getSwapTxsReceiptAnalyticsProperties(
   )
 
   const approveTx = trackedTxs.length > 1 ? txs[0] : undefined
-  const swapTx = txs[txs.length - 1]
+  const swapTx = trackedTxs.length > 0 ? txs[txs.length - 1] : undefined
```

Comment:
do we need these changes?

Reply:
yes this is what caused the initial crash, `swapTx` could be undefined, but TS didn't know it.